src/berry_pouch.c

```diff
@@ -0,0 +1,1528 @@
+#include "item.h"
+#include "item_use.h"
+#include "party_menu.h"
+#include "text_window.h"
+#include "strings.h"
+#include "string_util.h"
+#include "sound.h"
+#include "link.h"
+#include "money.h"
+#include "shop.h"
+#include "menu.h"
+#include "menu_indicators.h"
+#include "pokemon_storage_system.h"
+#include "constants/items.h"
+#include "constants/songs.h"
+
+struct BerryPouchStruct_203F36C
+{
+    void (*exitCallback)(void);
+    u16 indicatorOffset;
+    u8 indicatorTaskId;
+    u8 listMenuNumItems;
+    u8 listMenuMaxShowed;
+    u8 itemMenuIconId;
+    u8 ALIGNED(4) bg1TilemapBuffer[BG_SCREEN_SIZE];
+    s16 data[4];
+};
+
+struct BerryPouchStruct_203F370
+{
+    void (*savedCallback)(void);
+    u8 type;
+    u8 allowSelect;
+    u8 unused_06;
+    u16 listMenuSelectedRow;
+    u16 listMenuScrollOffset;
+};
+
+enum
+{
+    BP_ACTION_USE = 0,
+    BP_ACTION_TOSS,
+    BP_ACTION_GIVE,
+    BP_ACTION_EXIT,
+    BP_ACTION_DUMMY
+};
+
+static EWRAM_DATA struct BerryPouchStruct_203F36C *sResources = NULL;
+static EWRAM_DATA struct BerryPouchStruct_203F370 sStaticCnt = {};
+static EWRAM_DATA struct ListMenuItem *sListMenuItems = NULL;
+static EWRAM_DATA u8 *sListMenuStrbuf = NULL;
+static EWRAM_DATA const u8 *sContextMenuOptions = NULL;
+static EWRAM_DATA u8 sContextMenuNumOptions = 0;
+static ALIGNED(4) EWRAM_DATA u8 sVariableWindowIds[14] = {};
+static ALIGNED(4) EWRAM_DATA u8 sBerryPouchSpriteId = 0;
+
+static void CB2_InitBerryPouch(void);
+static bool8 RunBerryPouchInit(void);
+static void AbortBerryPouchLoading(void);
+static void Task_AbortBerryPouchLoading_WaitFade(u8 taskId);
+static void BerryPouchInitBgs(void);
+static bool8 BerryPouchLoadGfx(void);
+static bool8 AllocateListMenuBuffers(void);
+static void SetUpListMenuTemplate(void);
+static void GetBerryNameAndIndexForMenu(u8 * dest, u16 itemId);
+static void BerryPouchMoveCursorFunc(s32 itemIndex, bool8 onInit, struct ListMenu *list);
+static void BerryPouchItemPrintFunc(u8 windowId, s32 itemId, u8 y);
+static void BerryPouchSetArrowCursorAt(u8 y, u8 colorIdx);
+static void PrintSelectedBerryDescription(s32 itemIndex);
+static void CreateScrollIndicatorArrows_BerryPouchList(void);
+static void PrintBerryPouchHeaderCentered(void);
+static void SanitizeListMenuSelectionParams(void);
+static void UpdateListMenuScrollOffset(void);
+static void BerryPouch_DestroyResources(void);
+static void Task_BerryPouchFadeToExitCallback(u8 taskId);
+static void SortAndCountBerries(void);
+static void Task_BerryPouchMain(u8 taskId);
+static void Task_NormalContextMenu_0(u8 taskId);
+static void Task_NormalContextMenu_1(u8 taskId);
+static void Task_BerryPouch_Use_0(u8 taskId);
+static void Task_BerryPouch_Toss_0(u8 taskId);
+static void Task_AskTossMultiple(u8 taskId);
+static void Task_TossNo(u8 taskId);
+static void Task_Toss_SelectMultiple(u8 taskId);
+static void Task_TossYes(u8 taskId);
+static void Task_WaitButtonThenTossBerries(u8 taskId);
+static void Task_BerryPouch_Give_0(u8 taskId);
```

Comment:
Why do we need this `_0`(for multiple functions)?

Reply:
Was meant to be intermediate nomenclature. Resolved.